src/mc_rbdyn/Robot.cpp

```diff
@@ -740,6 +745,26 @@ const ForceSensor & Robot::bodyForceSensor(const std::string & body) const
   return forceSensors_.at(bodyForceSensors_.at(body));
 }
 
+const ForceSensor & Robot::findBodyForceSensor(const std::string & body) const
+{
+  unsigned nextIndex = bodyIndexByName(body);
+  while(nextIndex != 0)
```

Comment:
`nextIndex` should be `int` (as returned by `mb().parent(...)`) and the condition should be `nextIndex >= 0`